pkg/utils/parse_arf_result.go

```diff
@@ -17,30 +17,34 @@ const (
 	machineConfigFixType = "urn:xccdf:fix:script:ignition"
 )
 
-func ParseRemediationsFromArf(scheme *runtime.Scheme, scanName string, namespace string, arf string) ([]*complianceoperatorv1alpha1.ComplianceRemediation, error) {
+func ParseRemediationFromContentAndResults(scheme *runtime.Scheme, scanName string, namespace string, content, results string) ([]*complianceoperatorv1alpha1.ComplianceRemediation, error) {
 	remediations := make([]*complianceoperatorv1alpha1.ComplianceRemediation, 0)
 
-	reader := strings.NewReader(arf)
-	dom, err := xmldom.Parse(reader)
+	resultsReader := strings.NewReader(results)
```

Comment:
we wouldn't need this function if we would pass the reader directly. We could just pass this as the Reader interface, and not load the results in memory in the function that calls this.

Reply:
Fixed locally